lib/babe/verify.go

```diff
@@ -331,13 +332,49 @@ func (b *verifier) verifyAuthorshipRight(header *types.Header) error {
 	if err != nil {
 		return fmt.Errorf("could not verify block equivocation: %w", err)
 	}
+
 	if equivocated {
 		return fmt.Errorf("%w for block header %s", ErrProducerEquivocated, header.Hash())
 	}
 
 	return nil
 }
 
+func (b *verifier) submitAndReportEquivocation(
+	slot uint64, authorityIndex uint32, firstHeader, secondHeader types.Header) error {
+
+	// TODO: Check if it is initial sync
+	// don't report any equivocations during initial sync
+	// as they are most likely stale.
```

Comment:
I will create a separate issue for this. substrate has something called `BlockOrigin`, which tells us how did we get this block. It could be things like we got it in the initial sync, was broadcasted in the network, genesis block, got it from a file, etc.

I think adding this initial sync check could be its own PR. It mostly isn't unsafe to not have this check for now.

Reply:
created issue here https://github.com/ChainSafe/gossamer/issues/3004